src/tests/tracing/eventcounter/runtimecounters.cs

```diff
@@ -104,7 +104,7 @@ public static int TestEntryPoint()
             // Create an EventListener.
             using (RuntimeCounterListener myListener = new RuntimeCounterListener())
             {
-                Thread.Sleep(3000);
+                Thread.Sleep(60000);
```

Comment:
A 1 minute sleep in an inner loop test seems like too much. Is there some to wait exactly until the counters are ready?

Reply:
What do you think about smth like this?
```c#
            // Create an EventListener.
            using (RuntimeCounterListener myListener = new RuntimeCounterListener())
            {
                // Wait max 60 seconds
                for (int i = 0; i < 60; i++)
                {
                    Thread.Sleep(1000);
                    if (myListener.Verify())
                    {
                        Console.WriteLine("Test passed");
                        return 100;
                    }
                }

                Console.WriteLine($"Test Failed - did not see one or more of the expected runtime counters.");
                return 1;
            }
```

Reply:
Checked version from https://github.com/dotnet/runtime/pull/104461#discussion_r1666536809 with checked build on riscv64 VisionFive2:
```
Did not see cpu-usage
Saw cpu-usage
Saw working-set
Saw gc-heap-size
Saw gen-0-gc-count
Saw gen-1-gc-count
Saw gen-2-gc-count
Saw threadpool-thread-count
Saw monitor-lock-contention-count
Saw threadpool-queue-length
Saw threadpool-completed-items-count
Saw alloc-rate
Saw active-timer-count
Saw gc-fragmentation
Saw gc-committed
Saw exception-count
Saw time-in-gc
Saw gen-0-size
Saw gen-1-size
Saw gen-2-size
Saw loh-size
Saw poh-size
Saw assembly-count
Saw il-bytes-jitted
Saw methods-jitted-count
Saw time-in-jit
Saw gen-0-gc-budget
Saw total-pause-time-by-gc
Test passed
```

If you think that approach from https://github.com/dotnet/runtime/pull/104461#discussion_r1666536809 is ok, I'll update PR.

Reply:
I'm not familiar with this part of the runtime... we'll probably have to wait for @tommcdon once the US holiday is over.

Reply:
I do not have objections to the approach listed in https://github.com/dotnet/runtime/pull/104461#discussion_r1666536809.  Adding @noahfalk @hoyosjs for additional thoughts.

Reply:
Thanks, I've updated PR